src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -938,7 +938,8 @@
    csr :- CertificateRequest
    {:keys [cacert cakey signeddir ca-ttl serial cert-inventory]} :- CaSettings]
   (let [validity    (cert-validity-dates ca-ttl)
-        cacert      (utils/pem->cert cacert)
+        ;; if part of a CA bundle, the intermediate CA will be first in the chain
+        cacert      (first (utils/pem->certs cacert))
         signed-cert (utils/sign-certificate (utils/get-subject-from-x509-certificate
                                              cacert)
                                             (utils/pem->private-key cakey)
```

Comment:
We may want to update our terminology to ensure we're not talking about root "upstream" cert/key.

We may also want to check the leaf cert that we pull from the cert chain matches the private key that we read from disk and error if they do not match.